src/Microsoft.AspNetCore.Authentication.OAuth/OAuthHandler.cs

```diff
@@ -177,7 +172,12 @@ public OAuthHandler(HttpClient backchannel)
                 throw new ArgumentNullException(nameof(context));
             }
 
-            var properties = new AuthenticationProperties(context.Properties);
+            var properties = new AuthenticationProperties(context.Properties)
+            {
+                // Review: should we add an option to configure the state lifetime?
+                ExpiresUtc = Options.SystemClock.UtcNow.AddMinutes(15)
```

Comment:
The property name is rather clear (though maybe not specific enough?), but I'm not sure we should make it nullable (which would result in a session cookie). At least, not without a warning indicating that setting it to null may result in a terrible cookie flooding :smile: